pkg/component/operator/web/v0/crawl_website.go

```diff
@@ -198,6 +209,33 @@ func (e *execution) CrawlWebsite(input *structpb.Struct) (*structpb.Struct, erro
 		c.Wait()
 	}()
 
+	// To avoid to wait for 2 minutes, we use a timer to check if there is a new page.
+	// If there is no new page, we cancel the context.
+	go func() {
+		inactivityTimer := time.NewTimer(2 * time.Second)
+		defer inactivityTimer.Stop()
+
+		for {
+			select {
+			case <-pageUpdateCh:
+				// Reset the timer whenever we get a new page
+				if !inactivityTimer.Stop() {
+					select {
+					case <-inactivityTimer.C:
+					default:
+					}
```

Comment:
Again, why a select statement with a default instead of reading the channel? Can't you just read the timer? Also, why do we need to `Stop` instead of just `Reset`ting the timer?